gen_host_status.yml

```diff
@@ -0,0 +1,36 @@
+- hosts: all
+  gather_facts: false
+  tasks:
+    - name: A debug msg all hosts will show except for skipped ones
+      debug:
+        msg: "Playing {{ ansible_host }}"
+      when: "'_skipped' not in ansible_host"
+
+    - name: Hosts haven't really changed, but we will say they have
+      debug:
+        msg: "I am a changed host."
+      changed_when: "'_changed' in ansible_host"
+      when: "'_changed' in ansible_host"
+
+    - name: All failhosts aboard the failboat
+      fail:
+        msg: "I did nothing to deserve this."
+      when: "'_failed' in ansible_host"
+
+    - name: Ignore this failure for some hosts
+      fail:
+        msg: "<insert inspirational quote about failure>"
+      ignore_errors: true
+      when: "'_ignored' in ansible_host"
+
+    - name: Fail and rescue - collection of tasks
+      block:
+        - fail:
+            msg: "HALP!!!"
+          when: "'_rescued' in ansible_host"
+      rescue:
+        - debug: msg="ε-(´・｀) ﾌ"
+
+    - name: Reach out to the unreachable hosts
+      ping:
+      when: "'_unreachable' in ansible_host"
```

Comment:
This seems to rely on the hostname being unreachable. I guess that's probably a safe assumption, but depending on the host you could accidentally hit a real server somewhere out there. That's why I think it might be good to include this:

```yaml
      vars:
        ansible_host: 'invalid.invalid'
```

As @ryanpetrello taught me, this is guaranteed to fail locally, and should avoid unanticipated network side effects or security problems.

Reply:
`example.org` is also a thing that exists explicitly for this reason, and `unreachable.example.org` will fail to resolve.

Reply:
I tested the `invalid.invalid`. But seems like the `vars` takes effect before `when` and then `'_unreachable' in ansible_host` won't be matched.

```
TASK [Reach out to the unreachable hosts] **************************************
task path: /projects/_520__project_musicshoot/gen_host_status.yml:34
skipping: [6_rescued] => {
    "changed": false, 
    "skip_reason": "Conditional result was False"
}
skipping: [5_ignored] => {
    "changed": false, 
    "skip_reason": "Conditional result was False"
}
skipping: [1] => {
    "changed": false, 
    "skip_reason": "Conditional result was False"
}
skipping: [2_skipped] => {
    "changed": false, 
    "skip_reason": "Conditional result was False"
}
skipping: [7_unreachable] => {
    "changed": false, 
    "skip_reason": "Conditional result was False"
}
skipping: [3_changed] => {
    "changed": false, 
    "skip_reason": "Conditional result was False"
}
```

Reply:
you could work around that by first doing `set_fact` and then doing that task. I know that's inelegant, but that kind of hack is not uncommon.